packages/angular_devkit/schematics/src/tree/host-tree_spec.ts

```diff
@@ -7,9 +7,38 @@
  */
 import { normalize, virtualFs } from '@angular-devkit/core';
 import { FilterHostTree, HostTree } from './host-tree';
+import { MergeStrategy } from './interface';
 
 describe('HostTree', () => {
+  describe('merge', () => {
+    it('should create files from each tree', () => {
+      const tree = new HostTree();
+      tree.create('/file1', '');
+      const tree2 = new HostTree();
+      tree2.create('/file2', '');
+      tree.merge(tree2);
+      expect(tree.actions[0].kind).toEqual('c');
+      expect(tree.actions[1].kind).toEqual('c');
+    });
+
+    it('should overwrite if the file exists in one tree', () => {
+      const tree = new HostTree();
+      tree.create('/file1', '');
+      const tree2 = new HostTree();
+      tree2.create('/file1', '');
+      tree.merge(tree2, MergeStrategy.Overwrite);
+      expect(tree.actions[0].kind).toEqual('c');
+      expect(tree.actions[1].kind).toEqual('o');
```

Comment:
Since the file content is the same, there should only be a single create action.
Can you also add a test with different content?